src/jit_x64.c

```diff
@@ -0,0 +1,594 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+
+/* This JIT implementation has undergone extensive modifications, heavily
+ * relying on the ubpf_jit_x86_64.[ch] from ubpf. The original
+ * ubpf_jit_x86_64.[ch] file served as the foundation and source of inspiration
+ * for adapting and tailoring it specifically for this JIT implementation.
+ * Therefore, credit and sincere thanks are extended to ubpf for their
+ * invaluable work.
+ * Reference:
+ *   https://github.com/iovisor/ubpf/blob/main/vm/ubpf_jit_x86_64.c
+ */
+
```

Comment:
Add compilation guard. That is,
```c
#if !defined(__x86_64__)
#error "This implementation is dedicated to x86-64."
#endif
```

See https://github.com/sysprog21/kvm-host/blob/master/src/arch/x86/vm.c as an example.